spring-boot-project/spring-boot-test-autoconfigure/src/test/java/org/springframework/boot/test/autoconfigure/web/client/RestClientTestNoComponentIntegrationTests.java

```diff
@@ -62,8 +62,9 @@ void examplePropertiesIsNotInjected() {
 	@Test
 	void manuallyCreateBean() {
 		ExampleRestClient client = new ExampleRestClient(this.restTemplateBuilder);
-		this.server.expect(requestTo("/test")).andRespond(withSuccess("hello", MediaType.TEXT_HTML));
+		this.customizer.getServer(client.getRestTemplate()).expect(requestTo("/test")).andRespond(withSuccess("hello", MediaType.TEXT_HTML));
 		assertThat(client.test()).isEqualTo("hello");
+		this.customizer.getServer(client.getRestTemplate()).reset();
```

Comment:
Can you not just call something like ```this.server.reset()```?

Reply:
I tried `this.server.reset()` at first, but this did not work. Actually, for [MockRestServiceServer](https://www.javadoc.io/doc/org.springframework/spring-test/4.2.8.RELEASE/org/springframework/test/web/client/MockRestServiceServer.html), there is no `reset` API. The` reset` is for `MockServerRestTemplateCustomizer` to remove RestTemplate.

The error message is `Unable to use auto-configured MockRestServiceServer since MockServerRestTemplateCustomizer has been bound to more than one RestTemplate.` This is because if run the tests multiple times, there will be more than one client, but the server was already bound to the RestTemplate from the first client. So I want to remove or unbound the RestTemplate. I wanted to find some API to unbind the server, but I didn’t find it. 

I checked [MockRestServiceServerAutoConfiguration](https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot-test-autoconfigure/src/main/java/org/springframework/boot/test/autoconfigure/web/client/MockRestServiceServerAutoConfiguration.java),  we can reset the internal state and remove all expectations and recorded requests by `reset`. This test doesn’t use auto-configuration to set the server. So I think it's better to call the customizer to auto-config the server each time a new client runs. And since the customizer should be used more than once,  we can use `getServer(RestTemplate)` to access the related server.

Reply:
Thanks for the information. The fix looks fine to me now. 

This fix looked a bit unusual because it is changing the class that the test is exercising. However, after skimming the new class in this fix, it appears that the new class is mainly a wrapper for the old one.